terraform/main.tf

```diff
@@ -17,8 +17,8 @@ module "hello-world" {
   name               = "hello-world"
   path               = "hello-world"
   apigee_environment = var.apigee_environment
-  proxy_type         = "sandbox"
+  proxy_type         = (var.force_sandbox || length(regexall("sandbox", var.apigee_environment)) > 0) ? "sandbox" : "live"
   namespace          = var.namespace
-  make_api_product   = length(var.namespace) == 0
-  api_product_display_name = "Hello World Api"
+  make_api_product   = !(length(regexall("sandbox", var.apigee_environment)) > 0)
```

Comment:
I think this can be simplified to `length(regexall("sandbox", var.apigee_environment)) == 0` Also I think we don't want product if namespace is defined. So:
 `length(regexall("sandbox", var.apigee_environment)) == 0 && length(var.namespace) == 0`